app/src/main/java/de/vectordata/skynet/crypto/keys/KeyProvider.java

```diff
@@ -1,9 +1,7 @@
 package de.vectordata.skynet.crypto.keys;
 
-import de.vectordata.skynet.net.packet.P0BChannelMessage;
-
 public interface KeyProvider {
 
-    ChannelKeys getChannelKeys(P0BChannelMessage message);
+    ChannelKeys getChannelKeys(long channelId);
```

Comment:
One channel can have multiple keys during its lifetime. Even for sending a message this API might be not suitable because a new public key could be uploaded by the counterpart in the meantime rendering the referenced `DirectChannelUpdate` and actually used keypair to be different.

Reply:
Currently, the latest key is returned for this. In the future this will be changed to optionally also take in a message ID so that it can return the latest key.

Reply:
This can be done later and will be tracked by #16.